File: desert/medical-attention.lua
-- using "bash", "cleave"
loop
    if 3 * self.health < self.maxHealth then
        self:moveXY(65, 46)
    else
        local e = self:findNearestEnemy()
        if e then
            local d = self:distanceTo(e)
            if e.health > 66 and self:isReady("bash") then
                self:bash(e)
            elseif d < 10 and d > 3 and self:isReady("cleave") then
                self:cleave(e)
            else
                self:attack(e)
            end
        end
    end
end

File: desert/medical-attention.clj
(while true
  (def e (.findNearest this (.findEnemies this)))
  (if e
    (if (and (> (.health e) 78) (.isReady this "bash"))
      (.bash this e)
      (.attack this e))
    (if (< (.health this) (.maxHealth this))
      (when true
        (.move this (clj->js { :x 65, :y 46 }))
        (if (and (= (.x (.pos this)) 65) (= (.y (.pos this)) 46))
          (.say this "heal me")))
      (.move this (clj->js { :x 51, :y 38 })))))
